lib/cocoapods/generator/embed_frameworks_script.rb

```diff
@@ -129,27 +129,29 @@ def script
           # Copies and strips a vendored dSYM
           install_dsym() {
             local source="$1"
+            warn_missing_arch=${2:-true}
             if [ -r "$source" ]; then
-              # Copy the dSYM into a the targets temp dir.
+              # Copy the dSYM into the targets temp dir.
               echo "rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --filter \\"- CVS/\\" --filter \\"- .svn/\\" --filter \\"- .git/\\" --filter \\"- .hg/\\" --filter \\"- Headers\\" --filter \\"- PrivateHeaders\\" --filter \\"- Modules\\" \\"${source}\\" \\"${DERIVED_FILES_DIR}\\""
               rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "${source}" "${DERIVED_FILES_DIR}"
 
               local basename
-              basename="$(basename -s .framework.dSYM "$source")"
-              binary="${DERIVED_FILES_DIR}/${basename}.framework.dSYM/Contents/Resources/DWARF/${basename}"
+              basename="$(basename -s .dSYM "$source")"
+              binary_name="$(ls "$source/Contents/Resources/DWARF")"
+              binary="${DERIVED_FILES_DIR}/${basename}.dSYM/Contents/Resources/DWARF/${binary_name}"
 
               # Strip invalid architectures so "fat" simulator / device frameworks work on device
               if [[ "$(file "$binary")" == *"Mach-O "*"dSYM companion"* ]]; then
-                strip_invalid_archs "$binary"
+                strip_invalid_archs "$binary" "$warn_missing_arch"
               fi
 
               if [[ $STRIP_BINARY_RETVAL == 1 ]]; then
                 # Move the stripped file into its final destination.
                 echo "rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --links --filter \\"- CVS/\\" --filter \\"- .svn/\\" --filter \\"- .git/\\" --filter \\"- .hg/\\" --filter \\"- Headers\\" --filter \\"- PrivateHeaders\\" --filter \\"- Modules\\" \\"${DERIVED_FILES_DIR}/${basename}.framework.dSYM\\" \\"${DWARF_DSYM_FOLDER_PATH}\\""
-                rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --links --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "${DERIVED_FILES_DIR}/${basename}.framework.dSYM" "${DWARF_DSYM_FOLDER_PATH}"
+                rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --links --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "${DERIVED_FILES_DIR}/${basename}.dSYM" "${DWARF_DSYM_FOLDER_PATH}"
               else
                 # The dSYM was not stripped at all, in this case touch a fake folder so the input/output paths from Xcode do not reexecute this script because the file is missing.
-                touch "${DWARF_DSYM_FOLDER_PATH}/${basename}.framework.dSYM"
```

Comment:
this was the expected format for vendored frameworks, but since xcframeworks will have many dSYM files they must be named differently